### PR TITLE
Add support for rate control for 1000+ conns/s

### PR DESCRIPTION
**Pull request by qzhuyan**
With only arg '-i' (interval), one could not achieve conn rate > 1000 connections/s because the interval is in ms.
This PR add new arg "-R" to specify the desired connection rate (conns per secs)

- add `-R` for rate control
- new counters for success/fail connection attempts.
- Fix a bug in stats printing. 
- Smart select number of workers.

### CODE REVIEW

File: src/emqtt_bench.erl [hunk 1]
```diff
@@ -94,7 +94,8 @@
           "minimum randomized period in ms that each publisher will wait before "
           "starting to publish (uniform distribution)"},
          {num_retry_connect, undefined, "num-retry-connect", {integer, 0},
-          "number of times to retry estabilishing a connection before giving up"}
+          "number of times to retry estabilishing a connection before giving up"},
+         {conn_rate, $R, "connrate", {integer, 0}, "connection rate(/s), default: -1, fallback to use --interval"}
```
Comment (thalesmg) [inline src/emqtt_bench.erl:98]:
```suggestion
         {conn_rate, $R, "connrate", {integer, 0}, "connection rate(/s), default: 0, fallback to use --interval"}
```
?

File: src/emqtt_bench.erl [hunk 2]
```diff
@@ -144,7 +145,8 @@
           "use short ids for client ids"},
          {lowmem, $l, "lowmem", boolean, "enable low mem mode, but use more CPU"},
          {num_retry_connect, undefined, "num-retry-connect", {integer, 0},
-          "number of times to retry estabilishing a connection before giving up"}
+          "number of times to retry estabilishing a connection before giving up"},
+         {conn_rate, $R, "connrate", {integer, 0}, "connection rate(/s), default: -1, fallback to use --interval"}
```
Comment (thalesmg) [inline src/emqtt_bench.erl:149]:
```suggestion
         {conn_rate, $R, "connrate", {integer, 0}, "connection rate(/s), default: 0, fallback to use --interval"}
```
?

File: src/emqtt_bench.erl [hunk 3]
```diff
@@ -188,7 +190,8 @@
...
```
Comment (thalesmg) [inline src/emqtt_bench.erl:194]:
```suggestion
         {conn_rate, $R, "connrate", {integer, 0}, "connection rate(/s), default: 0, fallback to use --interval"}
```
?

**Review by thalesmg — APPROVED** [review]
Awesome!  This will be a lot more convenient to use!  :beers:

File: src/emqtt_bench.erl (hunk 3 above)
Comment (qzhuyan) [reply src/emqtt_bench.erl]: yes, thanks.

**Pull request merged**